server/controllers/auth/login.js

```diff
@@ -0,0 +1,85 @@
+// modules
+const bcrypt = require('bcrypt');
+const jwt = require('jsonwebtoken');
+
+const database = require("../../database");
+
+exports.login = (req, res, next) => {
+  // todo: 'remember me' feature
+  const email = req.body.email;
+  const password = req.body.password;
+
+  // find member using email address
+  database.query(`
+    SELECT
+      member_id, first_name, last_name, email_address, password, profile_picture, is_owner, is_moderator, is_blocked, created_at
+    FROM
+      member
+    WHERE
+      email_address = '${email}'
+  ;`).then(databaseResult => {
+    // get first member from array of objects
+    const memberData = databaseResult.rows[0];
+
+    // validate member password with encrypted password in database
+    bcrypt.compare(password, memberData.password).then(validatePassword => {
```

Comment:
This has two level nested `then`. `async/await` would be better here.

Reply:
🤔